llvm/utils/update_mc_test_checks.py

```diff
@@ -298,23 +316,58 @@ def main():
                     else:
                         gen_prefix += getStdCheckLine(prefix, o, mc_mode)
 
-            generated_prefixes.append(gen_prefix.rstrip("\n"))
+            generated_prefixes[input_line] = gen_prefix.rstrip("\n")
 
         # write output
-        prefix_id = 0
+        written_lines = set()
         for input_info in ti.iterlines(output_lines):
             input_line = input_info.line
-            if isTestLine(input_line, mc_mode):
+            if input_line in testlines:
+                if ti.args.remove_duplicate:
+                    if input_line in written_lines:
```

Comment:
So does this mean we actually only allow unique lines in the output with this option? How will this work with multiple cases having identical comments?

I think this option should aim to remove identical test cases rather than individual lines? Probably with duplicates reported regardless of whether this option is used?

Reply:
It only remove duplicated test line. If there are same comments they should stay.

I can add a case for this in the test

Reply:
added.

duplicated comment should not be dropped